### PR TITLE
Promote dataset

### PR DESCRIPTION
**Pull request by ioumu8**

### CODE REVIEW

File: dataset.py [hunk 1]
```diff
@@ -38,3 +38,14 @@ def append(self, x):
 
     def extend(self, iterable):
         list.extend(self, list(map(self.materialize, iterable)))
+
+class Dataset(dict):
+    """
+    Merge all data into one set.
+    This will be calle in Simulator
```
Comment (b821213) [inline dataset.py]: Typo in `calle`.

Comment (b821213) [reply dataset.py]: `in the simulator.`

File: get_data.py [hunk 2]
```diff
@@ -16,8 +16,8 @@ def call_crawler(data_list):
         old_data = DataLoader([data_name]).get_raw_data()
         new_data = crawler.update()
         with open(data_name, 'wb') as file:
-            for key in new_data:
+            for key in new_data[data_name]:
                 if key in old_data[data_name].keys():
-                    new_data[key] = old_data[data_name][key] + new_data[key]
+                    new_data[data_name][key] = old_data[data_name][key] + new_data[data_name][key]
```
Comment (ioumu8) [inline get_data.py]: It is amazing that, I could use the method of Dataset without import it.

Comment (ioumu8) [reply get_data.py]: When I tested it, it really worked.

File: dataset.py [hunk 3]
```diff
@@ -38,3 +38,15 @@ def append(self, x):
...
+    """
+    Merge all data into one set.
+    This will be called in the simulator.
+    """
+    def __init__(self, data_name, instruments, data):
+        dict.__init__(self)
+        self.data_name = data_name
+        self[data_name] = {}
+        for instrument in instruments:
+            self[data_name][instrument] = data[data_name][instrument]
```
Comment (b821213) [inline dataset.py]: Use init mentioned above can be clearer.

File: dataset.py [hunk 4]
```diff
@@ -38,3 +38,15 @@ def append(self, x):
...
     def extend(self, iterable):
         list.extend(self, list(map(self.materialize, iterable)))
+
+class Dataset(dict):
+    """
+    Merge all data into one set.
+    This will be called in the simulator.
+    """
+    def __init__(self, data_name, instruments, data):
+        dict.__init__(self)
```
Comment (b821213) [inline dataset.py]: `dict.__init__(self, data)`

File: get_data.py (hunk 2 above)
Comment (b821213) [reply get_data.py]: > It is amazing that, I could use the method of Dataset without import it.

I think as long as you don't really "need" the name of the class, you don't need to import it.
That's what duck typing wants.

File: dataset.py [hunk 5]
```diff
@@ -38,3 +38,15 @@ def append(self, x):
 
     def extend(self, iterable):
         list.extend(self, list(map(self.materialize, iterable)))
+
+class Dataset(dict):
+    """
+    Merge all data into one set.
+    This will be called in the simulator.
+    """
+    def __init__(self, data_name, instruments, data):
```
Comment (b821213) [inline dataset.py]: This is quite weird. `instruments` should already be covered by `data`?

File: workdays.py [hunk 6]
```diff
@@ -59,4 +60,4 @@ def update(self):
         return_data = {}
         for region in self.regions:
             return_data[region] = self.set_workdays(region)
-        return return_data
+        return Dataset(self.data_name, self.regions, {self.data_name : return_data})
```
Comment (b821213) [inline workdays.py]: Why not just `Dataset(self.data_name, return_data)`?

File: workdays.py [hunk 7]
```diff
@@ -5,6 +5,7 @@
 
 import os
 import pickle
+from dataset import Data
 from dataset import Dataset
```
Comment (b821213) [inline workdays.py]: `from dataset import Data, Dataset`

File: get_data.py [hunk 8]
```diff
@@ -16,8 +16,8 @@ def call_crawler(data_list):
         old_data = DataLoader([data_name]).get_raw_data()
         new_data = crawler.update()
         with open(data_name, 'wb') as file:
-            for key in new_data:
+            for key in new_data[data_name]:
```
Comment (b821213) [inline get_data.py]: The loader is already for `data_name`. I don't know why we must specify it again?

**Pull request merged**